projects/packages/forms/src/blocks/form-progress-indicator/edit.js

```diff
@@ -1,31 +1,75 @@
 import { useBlockProps } from '@wordpress/block-editor';
+import { SVG, Path } from '@wordpress/components';
 import StepControls from '../shared/components/form-step-controls';
 import useParentFormClientId from '../shared/hooks/use-parent-form-client-id';
 import useStepNavigation from '../shared/hooks/use-step-navigation';
+import { calculateProgressPercentage } from '../shared/util/progress-calculation';
 
 import './editor.scss';
 
 const FormProgressIndicatorEdit = ( { clientId } ) => {
 	const parentFormId = useParentFormClientId( clientId );
 	const { currentStepInfo, steps } = useStepNavigation( parentFormId );
 
-	let progress = steps.length ? ( ( currentStepInfo.index + 1 ) / steps.length ) * 100 : 10;
-	if ( currentStepInfo.index === -1 && steps.length > 0 ) {
-		progress = ( 1 / steps.length ) * 100; // Assume the first step is active
-	}
-
 	const blockProps = useBlockProps();
+	const isDotStyle = blockProps.className && blockProps.className.includes( 'is-style-dots' );
 
-	// Only need to set width – colours come from core style engine variables.
-	const progressBarStyle = {
-		width: `${ progress }%`,
-	};
+	// Use shared progress calculation logic
+	const currentStep = currentStepInfo.index + 1;
+	let progressPercentage = calculateProgressPercentage( currentStep, steps.length, isDotStyle );
+
+	// Show 25% progress in "All steps" view for line style to preview the bar
+	if ( ! isDotStyle && currentStepInfo.index === -1 && steps.length > 0 ) {
+		progressPercentage = 25;
+	}
 
 	return (
 		<>
 			<div className="jetpack-form-progress-indicator--wrapper">
 				<div { ...blockProps }>
-					<div className="jetpack-form-progress-indicator-bar" style={ progressBarStyle }></div>
+					<div className="jetpack-form-progress-indicator-steps">
+						<div
+							className="jetpack-form-progress-indicator-progress"
+							style={ { width: `${ progressPercentage }%` } }
+						></div>
+						{ steps.map( ( step, index ) => {
+							const isActive = index === currentStepInfo.index;
+							const isCompleted = index < currentStepInfo.index;
+
+							return (
+								<div
+									key={ index }
+									className={ `jetpack-form-progress-indicator-step${
+										isActive ? ' is-active' : ''
+									}${ isCompleted ? ' is-completed' : '' }` }
```

Comment:
```suggestion
					className={ clsx( {
						'jetpack-form-progress-indicator-step',
						'is-active': isActive,
						'is-completed': isCompleted,
					} ) }
```


I like to use `clsx` for readability, but it's a small detail. You'll still need to import it in this file then.